File: backend/server/models/event.py
from pymongo import MongoClient


class Event:
    """Model for Event"""

    def __init__(
        self, connection, event_id, name, effect, namespace, candidate, start, end
    ):
        self.connection = connection
        self.event_id = event_id
        self.name = name
        self.effect = effect
        self.namespace = namespace
        self.candidate = candidate
        self.start = start
        self.end = end

    def serialize(self):
        return {
            "event_id": self.event_id,
            "name": self.name,
            "effect": self.effect,
            "namespace": self.namespace,
            "candidate": self.candidate,
            "start": self.start,
            "end": self.end,
        }

    def write(self):
        db = self.connection.capitol_tracker

        collection = db.events

        collection.insert_one(self.serialize())

        return self.event_id

File: events/Sources/App/Models/Event.swift
import Vapor
import Fluent
import FluentMySQLDriver

final class Event: Model, Content {
    // Name of the table or collection.
    static let schema = "event"

    // Unique identifier for this Planet.
    @ID(key: .id)
    var id: UUID?

    // The Planet's name.
    @Field(key: "effect")
    var effect: String
    
    @Field(key: "end")
    var end: String
    
    @Field(key: "name")
    var name: String
    
    @Field(key: "namespace")
    var namespace: String
    
    @Field(key: "start")
    var start: String

    // Creates a new, empty Planet.
    init() { }

    // Creates a new Planet with all properties set.
    init(id: UUID? = nil, name: String) {
        self.id = id
        self.name = name
    }
}


